genet/schedule_elements.py

```diff
@@ -259,6 +259,60 @@ def find_epsg(self):
                     return epsg
         return None
 
+    @abstractmethod
+    def service_attribute_data(self, keys: Union[list, str], index_name: str = None):
+        pass
+
+    @abstractmethod
+    def route_attribute_data(self, keys: Union[list, str], index_name: str = None):
+        pass
+
+    @abstractmethod
+    def stop_attribute_data(self, keys: Union[list, str], index_name: str = None):
+        pass
+
+    @abstractmethod
+    def trips_with_stops_to_dataframe(self, gtfs_day='19700101') -> pd.DataFrame:
+        pass
```

Comment:
All of these methods used to exist just for the schedule, putting them here in the base class `ScheduleElement` (as empty abstract methods) forces all the other classes that depend on it (`Route` and `Service`) to have a version of those methods too. These methods are actually not useful for this PR (oops) but at I thought they were... and having done the work to move them out, I decided to leave it, I think it might some in handy to have them for routes and services 🤷 😄